frontend/reducers/addressBook.js

```diff
@@ -0,0 +1,46 @@
+import { RECEIVE_USER, SUCCESS_LOGOUT } from '@shopgate/pwa-common/constants/ActionTypes';
+import { SET_DEFAULT_ADDRESS } from './../constants/ActionTypes';
+import config from './../config';
+
+const { splitDefaultAddressesByTags = [] } = config;
+
+/**
+ * Stores the user addresses
+ * @param {Object} state The current user state
+ * @param {Object} action The action object
+ * @return {Object} The new state
+ */
+export default (state = {}, action) => {
+  switch (action.type) {
+    case RECEIVE_USER: {
+      const { addresses } = action.user;
+
+      const defaultAddresses = {};
+      splitDefaultAddressesByTags.forEach((tag) => {
```

Comment:
This seems to break when no address is returned it results into:
```
default: { default: {} }
```

Reply:
It was designed so. Since splitting to tags can be ["default"] || ["shipping", "billing"], etc
So main `default` is a wrapper for internal tag references, eg. 
```
default: { default: null|'long-if-of-address' }

default: { shipping: null|'long-if-of-address', billing:  null|'long-if-of-address'}
```